js/webserver.js

```diff
@@ -112,14 +112,10 @@ WebServer.prototype = {
           this.expressWsHttps = expressWs(app, this.httpsServer, {maxPayload: 50000});
           listening = true;
         } catch (e) {
-          if (e.message == 'mac verify failure' && !noPrompt) {
-            const r = reader();
-            try {
-              httpsOptions.passphrase = yield reader.readPassword(
+          if (e.message == 'mac verify failure') {
+            const r = new reader();
+            this.httpsOptions.passphrase = r.readPasswordSync(
```

Comment:
If you forget the password, this never ends (on windows anyway). I was unable to kill the server via control+c. I think it would be better to give up after 3 wrong password attempts. After 3 wrong password attempts, the server should quit with a message.